src/cgame/cg_rocket_draw.cpp

```diff
@@ -1545,18 +1543,21 @@ static void CG_ScanForCrosshairEntity()
 		return;
 	}
 
-	ownTeam = ( team_t ) cg.snap->ps.persistant[ PERS_TEAM ];
-	targetState = &cg_entities[ trace.entityNum ].currentState;
+	entityState_t &targetState = cg_entities[ trace.entityNum ].currentState;
+
+	team_t ownTeam    = CG_MyTeam();
+	team_t targetTeam = CG_Team(targetState);
+
+	if (!CG_IsAlive(targetState))
```

Comment:
I think it's only used when in a team playing, so that you should not have odd conditions where you don't have an entity.

Generally speaking if you are dead, you still have an entity and you do get an entity as soon as you spawn. I think you normally have an entity at all time?

Reply:
also, if you want to harden this function, I can add this in CG_MyTeam and other teams functions

```C
if (team <= TEAM_NONE || team >= NUM_TEAMS) return TEAM_NONE;
```

Reply:
No I mean the target entity. Maybe it could be a mover or something.

Reply:
I mean, if it's a mover then I'd be surprised if it's in a team, but yeah, that changed. The whole point of this code is to "tag" the crosshair depending if it's a friend or foe though.

Maybe one day we could make it react to destructable wall, etc, but those don't currently work in unv.

Reply:
I tried it out and you can indeed inspect non-alive entities. Hence the `!CG_IsAlive` check disrupts the functionality.
![unvanquished_2021-09-20_021849_000](https://user-images.githubusercontent.com/7809431/133968238-b0c4b98c-ed6c-4be4-97be-2d2453eae89d.jpg)
.